folding/src/standard_config.rs

```diff
@@ -0,0 +1,471 @@
+                    *a += b * challenge;
+                }
+            }
+            a
+        }
+    }
+    // and Witness
+    impl<G: KimchiCurve> Witness<G> for MyWitness<G> {}
+
+    // a type for the columns
+    #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+    enum MyCol {
+        A,
+        B,
+        C,
+    }
+
+    // implementing FoldingColumnTrait, trivial in this witness-only case
+    impl FoldingColumnTrait for MyCol {
+        fn is_witness(&self) -> bool {
+            true
+        }
+    }
+    // a challenge
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+    enum MyChallenge {
+        Beta,
+        Gamma,
+    }
+
+    // now, to use this config with our types, we need to implement Index
+    // if not already implemented, to resolve access to values in instance,
+    // witness, and structure if present
+
+    // for witness columns
+    impl<G: KimchiCurve> Index<MyCol> for MyWitness<G> {
+        type Output = Vec<G::ScalarField>;
+
+        fn index(&self, index: MyCol) -> &Self::Output {
+            let index = match index {
+                MyCol::A => 0,
+                MyCol::B => 1,
+                MyCol::C => 2,
+            };
+            &self.columns[index]
+        }
+    }
+    // for selectors, () in this case as we have none
+    impl<G: KimchiCurve> Index<()> for MyWitness<G> {
+        type Output = Vec<G::ScalarField>;
+
+        fn index(&self, _index: ()) -> &Self::Output {
+            unreachable!()
+        }
+    }
+    // for challenges, which should live in the instance
+    impl<G: KimchiCurve> Index<MyChallenge> for MyInstance<G> {
+        type Output = G::ScalarField;
+
+        fn index(&self, index: MyChallenge) -> &Self::Output {
+            match index {
+                MyChallenge::Beta => &self.beta,
+                MyChallenge::Gamma => &self.gamma,
+            }
+        }
+    }
+
+    // now we can get an instance of StandardConfig, where selectors and structures have
+    // default for cases like this where we don't need them
+    type MyConfig<G> = StandardConfig<G, MyCol, MyChallenge, MyInstance<G>, MyWitness<G>>;
+
+    // creating some example constraint
+    fn constraint<G: KimchiCurve>() -> FoldingCompatibleExpr<MyConfig<G>> {
+        let column = |col| {
+            FoldingCompatibleExpr::Atom(FoldingCompatibleExprInner::Cell(Variable {
+                col,
+                row: CurrOrNext::Curr,
+            }))
+        };
+        let chall =
+            |chall| FoldingCompatibleExpr::Atom(FoldingCompatibleExprInner::Challenge(chall));
+        let a = column(MyCol::A);
+        let b = column(MyCol::B);
+        let c = column(MyCol::C);
+        let beta = chall(MyChallenge::Beta);
+        let gamma = chall(MyChallenge::Gamma);
+        (a + b - c) * (beta + gamma)
+    }
+
+    // here an example of how the config would be used, which is similar to any
+    // other custom config.
+    #[allow(dead_code)]
```

Comment:
Maybe having this as a `test` and do a relatively “trivial” check?

Reply:
it doesn't work as a test, as I don't instance witness pairs to use as inputs, I made it just to show how it would be used and check there is no issue building it. But for a proper test I think it would be better to rewrite some of the ones we already have with this generic config.

Reply:
Agreed.